### PR TITLE
Add general interfaces for querybuilder plugin

### PR DESCRIPTION
**Pull request by Maxzor**
Some things needed, on top of the mainwindow exposition which is nice.
Added a method to rescroll to adjust to a list of items.

### CODE REVIEW

File: libobjrenderer/src/objectsscene.cpp [hunk 1]
```diff
@@ -854,6 +854,14 @@ QGraphicsView *ObjectsScene::getActiveViewport(void)
 	return(view_p);
 }
 
+void ObjectsScene::adjustViewportToItems(QList<BaseObjectView *> items)
```
Comment (rkhaotix) [inline libobjrenderer/src/objectsscene.cpp]: What's the purpose of this method? I don't get it.

File: libobjrenderer/src/objectsscene.cpp [hunk 2]
```diff
@@ -1433,6 +1441,14 @@ bool ObjectsScene::isMovingObjects(void)
 	return(moving_objs);
 }
 
+void ObjectsScene::selectItem(BaseObjectView * item)
```
Comment (rkhaotix) [inline libobjrenderer/src/objectsscene.cpp]: Why create a method that selects an item? All scene items has the setSelected that triggers the needed signals to handle item selection.

File: libpgmodeler_ui/src/mainwindow.cpp [hunk 3]
```diff
@@ -524,12 +524,33 @@ void MainWindow::restoreTemporaryModels(void)
 
 void MainWindow::showRightWidgetsBar(void)
 {
-	right_wgt_bar->setVisible(objects_btn->isChecked() || operations_btn->isChecked());
-}
+	int i=0;
+	bool show=false;
+	//This is currently the only way to enumerate widgets inside a layout...
+	//See https://stackoverflow.com/a/27225570/7359123
+	while(horizontalLayout_4->itemAt(i))
```
Comment (rkhaotix) [inline libpgmodeler_ui/src/mainwindow.cpp]: What do you mean by "enumerate"? Are you trying to toggle the visibility of children widgets in a layout?

File: libpgmodeler_ui/src/mainwindow.cpp [hunk 4]
```diff
@@ -2077,3 +2100,22 @@ void MainWindow::toggleLayersWidget(bool show)
 	layers_wgt->setVisible(show);
 }
 
+QPair <bool, SQLToolWidget *> MainWindow::isAnyManageDbOpened(void)
```
Comment (rkhaotix) [inline libpgmodeler_ui/src/mainwindow.cpp]: As far as I understood this method you need just to check if there's an instance of database being browsed, right? Why return a QPair and where do you need this data structure?

File: libpgmodeler_ui/src/modelwidget.h [hunk 5]
```diff
@@ -329,6 +329,12 @@ class ModelWidget: public QWidget {
 		//! \brief Returns the reference database model
 		DatabaseModel *getDatabaseModel(void);
 
+		//! \brief Returns the object scene used by the model
+		ObjectsScene *getObjectScene(void)const{return scene;};
```
Comment (rkhaotix) [inline libpgmodeler_ui/src/modelwidget.h]: Just a note... I like methods constructions in the .cpp file even if they have a single line of code. Just a matter of taste. You can leave this one like this... I can fix later.

File: libobjrenderer/src/objectsscene.cpp (hunk 1 above)
Comment (Maxzor) [reply libobjrenderer/src/objectsscene.cpp]: Check https://doc.qt.io/qt-5/qgraphicsview.html#fitInView
It both adjusts zoom and center, I figured it is nice to look at the query items.

File: libobjrenderer/src/objectsscene.cpp (hunk 2 above)
Comment (Maxzor) [reply libobjrenderer/src/objectsscene.cpp]: ahem... that is why I said "might not be needed" :fearful: :D
You can put it to garbage.

File: libpgmodeler_ui/src/mainwindow.cpp (hunk 3 above)
Comment (Maxzor) [reply libpgmodeler_ui/src/mainwindow.cpp]: Widgets are children of widgets, not layouts. So to find the buttons in a layout this is the only way I found.
Currently this function will display the bottom widget if one of the two buttons are checked.
I make it expandable so plugins can add an arbitrary number of dock widgets.
You made me realize there is an error : it should not check for the layer_btn.

File: libpgmodeler_ui/src/mainwindow.cpp (hunk 4 above)
Comment (Maxzor) [reply libpgmodeler_ui/src/mainwindow.cpp]: There must be a better way I know...
with the qpair I get a handle of the SQLToolWidget to talk to him directly :S

File: libpgmodeler_ui/src/modelwidget.h (hunk 5 above)
Comment (Maxzor) [reply libpgmodeler_ui/src/modelwidget.h]: Ok noted

**Pull request closed without merging**